### ISSUE TITLE
DataTable: cellNavigation with toggable columns

### ISSUE THREAD

**Issue by Rene-Stoeckel** [primefaces/primefaces#12195]
### Describe the bug

We are using the data tables with the new modern cellNavigation in our application. 
Some tables come with toggable columns (user can toggle the visibility of the column) where we observed the following effect:

When a column is toggled visible which was not visible before then its cells cannot be reached by the cell navigation via keyboard. (The cells which received a mouse click can be reached).

### Reproducer

I can provide a reproducer, but I guess the column toggle showcase for 14.0.2 would do as well.

### Expected behavior

All visible cells of the data table need to be reached via cell navigator / keyboard.
Maybe the internal structures of the cell navigator needs just to be updated when the visibility 
of a column is triggered.

### PrimeFaces edition

Community

### PrimeFaces version

14.0.2

### JSF implementation

MyFaces

### JSF version

2.3

### Java version

21

### Browser(s)

Chrome Version 126.0.6478.127 (Offizieller Build) (64-Bit), Microsoft Edge Version 126.0.2592.68 (Offizielles Build) (64-Bit)

**Comment by Rene-Stoeckel** [comment]
In this example the cursor can not reach the right column after it was toggled visible.
![image](https://github.com/primefaces/primefaces/assets/102955224/ab246c8f-8696-4dae-b4aa-24e044ca004a)

**Comment by melloware** [comment]
I can use the Showcase as a reproducer thanks!
